dashboard/src/app/workspaces/workspace-details/environments/environments.controller.js

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2015-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ */
+'use strict';
+
+/**
+ * @ngdoc controller
+ * @name workspace.details.controller:WorkspaceEnvironmentsController
+ * @description This class is handling the controller for details of workspace : section environments
+ * @author Oleksii Kurinnyi
+ */
+export class WorkspaceEnvironmentsController {
+
+  /**
+   * Default constructor that is using resource injection
+   * @ngInject for Dependency injection
+   */
+  constructor($q, $log, $scope, $timeout, cheEnvironmentRegistry) {
+    this.$q = $q;
+    this.$log = $log;
+    this.$scope = $scope;
+    this.cheEnvironmentRegistry = cheEnvironmentRegistry;
+
+    this.editorOptions = {
+      lineWrapping: true,
+      lineNumbers: false,
+      readOnly: true,
+      gutters: [],
+      onLoad: (editor) => {
+        $timeout(() => {
+          editor.refresh();
+        }, 1000);
+      }
+    };
+
+    this.init();
+  }
+
+  /**
+   * Sets initial values
+   */
+  init() {
+    this.newEnvironmentName = this.environmentName;
+    this.environment = this.workspaceConfig.environments[this.environmentName];
+
+    this.recipeType = this.environment.recipe.type;
+    this.environmentManager = this.cheEnvironmentRegistry.getEnvironmentManager(this.recipeType);
+
+    this.editorOptions.mode = this.environmentManager.editorMode;
+
+    this.machines = this.environmentManager.getMachines(this.environment);
+  }
+
+  /**
+   * Updates name of environment
+   * @param isFormValid {boolean}
+   */
+  updateEnvironmentName(isFormValid) {
+    if (!isFormValid || this.newEnvironmentName === this.environmentName) {
+      return;
+    }
+
+    this.workspaceConfig.environments[this.newEnvironmentName] = this.environment;
+    delete this.workspaceConfig.environments[this.environmentName];
+
+    if (this.workspaceConfig.defaultEnv === this.environmentName) {
+      this.workspaceConfig.defaultEnv = this.newEnvironmentName;
+    }
+
+    this.doUpdateEnvironments();
+  }
+
+  /**
+   * Callback which is called in order to update environment config
+   * @returns {Promise}
+   */
+  updateMachineConfig() {
```

Comment:
updateMachineConfig - rename to updateEnvironmentConfig then ("order to update environment config")